README.md
# laravel-DConverter
A helper tool for converting and formatting Outputs into human readable format.

#Integration in Laravel

##Install using Composer

Run `composer require sam0hack/dconverter  dev-master`

After you have installed dconverter , open your Laravel config file config/app.php and add the following lines in the $aliases array.

`'DConverter' => sam0hack\dconverter\DC::class,`

#Tools
**SizeFormat --- SizeFormat convert's the laravel getSize() Or bytes data into kb, mb and gb**
**Hours        --- Calculate Hours betweens Two dates**
**Percentage --- Calculate Percentage**
 **MakeColors --- Simply generate hex color**
**DistanceMatrix --- Calculate Time and distance between two locations using google distance API**

###Example of SizeFormat()

`DC::SizeFormat($attachment->file_size)`

#####In blade tamplate
`{{\sam0hack\dconverter\DC::SizeFormat($attachment->file_size)}}`

#####Output examples
* 99.24 kB
* 100 Mb
* 1 GB

###Example of Hours()

`DC::Hours($date,$date2)`

#####In blade tamplate
`{{\sam0hack\dconverter\DC::Hours($date,$date2)}}`

#####Output examples
* 240
* 1
* 60



###Example of Percentage()


    DC::Percentage($current,$total)

 calculate percentage using hours

    $totalhours = DC::Hours($start_date,$end_date);
    
    $currentHours = DC::Hours($start_date,$now_date);
    
    $percentage = DC::Percentage($currentHours,$totalhours);

#####Output examples
* 25
* 33.0
* -150


###Example of MakeColors()

`DC::MakeColors()`

#####In blade tamplate
`{{\sam0hack\dconverter\DC::MakeColors()}}`

#####Output examples
* "#0951e7"
* "#071377"
* "#cdda5d"


###Example of DistanceMatrix()
For Setup DistanceMatrix you need to add google API into config/services.php

'google'=>[
      'GOOGLE_API_KEY' => env('GOOGLE_MAP_API_KEY')
    ],
    OR without using env
 google'=>[
      'GOOGLE_API_KEY' => 'paste you google map api key here'
    ],   

> @Note Config key should be ***GOOGLE_API_KEY***

    DC::DistanceMatrix('New York Ave, El Paso, TX 79902, USA','86 W Houston St, New York, NY 10012, USA')
    $distance = $dist['distance'];
    $time	    = $dist['time'];

#####Output examples
* "2,184 mi"
*  "1 day 8 hours"

***DistanceMatrix Modes - Optionals***
For the calculation of distances, you may specify the transportation mode to use. By default, distances are 

 calculated for driving mode. The following travel modes are supported:
**driving** (default) indicates distance calculation using the road network.
**walking** requests distance calculation for walking via pedestrian paths & sidewalks (where available).
**bicycling** requests distance calculation for bicycling via bicycle paths & preferred streets (where available).

 

    DC::DistanceMatrix($origin, $destination,'walking');


#packagist
https://packagist.org/packages/sam0hack/dconverter





CHANGELOG.md
# Changelog

All Notable changes to `:FileSizeConverter` will be documented in this file.

Updates should follow the [Keep a CHANGELOG](http://keepachangelog.com/) principles.

## NEXT - YYYY-MM-DD

### Added
- SizeFormat

### Deprecated
- Nothing

### Fixed
- invalid syntax

### Removed
- Nothing

### Security
- Nothing


FileSizeConverter.php
<?php
/**
 * Created by PhpStorm.
 * User: sam0hack
 * Date: 4/11/16
 * Time: 4:28 AM
 */

namespace sam0hack;

/**
 * Class FileSizeConverter
 * @package FileSizeConverter
 */
class FileSizeConverter
{
    /**
     * @param $bytes
     * @return string
     */
    public static function SizeFormat($bytes)
    {
        if ($bytes >= 1073741824)
        {
            $bytes = number_format($bytes / 1073741824, 2) . ' GB';
        }
        elseif ($bytes >= 1048576)
        {
            $bytes = number_format($bytes / 1048576, 2) . ' MB';
        }
        elseif ($bytes >= 1024)
        {
            $bytes = number_format($bytes / 1024, 2) . ' kB';
        }
        elseif ($bytes > 1)
        {
            $bytes = $bytes . ' bytes';
        }
        elseif ($bytes == 1)
        {
            $bytes = $bytes . ' byte';
        }
        else
        {
            $bytes = '0 bytes';
        }

        return $bytes;
    }
}


src/FileSizeConverterServiceProvider.php
<?php

namespace sam0hack\FileSizeConverter;

use Illuminate\Support\ServiceProvider;

class FileSizeConverterServiceProvider extends ServiceProvider
{
    /**
     * Perform post-registration booting of services.
     *
     * @return void
     */
    public function boot()
    {
        echo "hello";
    }

    /**
     * Register any package services.
     *
     * @return void
     */
    public function register()
    {
        //
    }

    public static function SizeFormat($bytes)
    {
        if ($bytes >= 1073741824)
        {
            $bytes = number_format($bytes / 1073741824, 2) . ' GB';
        }
        elseif ($bytes >= 1048576)
        {
            $bytes = number_format($bytes / 1048576, 2) . ' MB';
        }
        elseif ($bytes >= 1024)
        {
            $bytes = number_format($bytes / 1024, 2) . ' kB';
        }
        elseif ($bytes > 1)
        {
            $bytes = $bytes . ' bytes';
        }
        elseif ($bytes == 1)
        {
            $bytes = $bytes . ' byte';
        }
        else
        {
            $bytes = '0 bytes';
        }
        return $bytes;
    }
}

src/DC.php
<?php
/**
 * Created by PhpStorm.
 * User: sam0hack
 * Date: 5/11/16
 * Time: 4:33 AM
 */

namespace sam0hack\dconverter;

/**
 * Class DC
 * @package sam0hack\dconverter
 */
class DC
{
    /**
     * @param $bytes
     * @return string
     */
    public static function SizeFormat($bytes)
    {
        if ($bytes >= 1073741824)
        {
            $bytes = number_format($bytes / 1073741824, 2) . ' GB';
        }
        elseif ($bytes >= 1048576)
        {
            $bytes = number_format($bytes / 1048576, 2) . ' MB';
        }
        elseif ($bytes >= 1024)
        {
            $bytes = number_format($bytes / 1024, 2) . ' kB';
        }
        elseif ($bytes > 1)
        {
            $bytes = $bytes . ' bytes';
        }
        elseif ($bytes == 1)
        {
            $bytes = $bytes . ' byte';
        }
        else
        {
            $bytes = '0 bytes';
        }
        return $bytes;
    }

    /**
     * @param $date1
     * @param $date2
     * @return float|int
     */
    public static function Hours($date1,$date2){

        $t1 = strtotime ( $date1 );
        $t2 = strtotime ( $date2 );
        $diff = $t2 - $t1;
        $hours = $diff / ( 60 * 60 );
        return $hours;
    }

    /**
     * @param $current
     * @param $total
     * @return float|int
     */
    public static function Percentage($current,$total){

        //1÷9×100
        $percentage = ($current / $total * 100);

        $percentage = number_format((float)$percentage, 2, '.', '');
        return $percentage;

    }

    /**
     * @return string
     */
    public static function random_color_part() {
    return str_pad( dechex( mt_rand( 0, 255 ) ), 2, '0', STR_PAD_LEFT);
}

    /**
     * @return string
     */
    public static function MakeColors(){
        $color = self::random_color_part().self::random_color_part().self::random_color_part();
        return $color;
    }


    /**
     * @param $origin
     * @param $destination
     * @param string $mode
     * @return array
     */
    public static function DistanceMatrix($origin, $destination,$mode='')
    {

        $app = app();

        $config = $app['config']['services.google'];

        $KEY = $config['GOOGLE_API_KEY'];


        if ($mode==''){
            $mode='driving';
        }


        $origin=urlencode($origin);
        $destination = urlencode($destination);
        $url = "https://maps.googleapis.com/maps/api/distancematrix/json?units=imperial&origins=$origin&destinations=$destination&mode=$mode&key=$KEY";

        $response_a = json_decode(file_get_contents($url), true);


        $dist = $response_a['rows'][0]['elements'][0]['distance']['text'];
        $time = $response_a['rows'][0]['elements'][0]['duration']['text'];


        return array('distance' => $dist, 'time' => $time);

    }



}

